### PR TITLE
Remove constant DRAGEN and replace usages for BCLCONVERT

### LINKED ISSUE #2030: Explicitly referring to DRAGEN software usage
## Description

In some instances we are using DRAGEN as a constant in the CG codebase. These include constants that should point to a more specific software that is run on dragen. One such example is the BclConverter class. Here we have the options "bcl2fastq" and "dragen". However, dragen runs the "bclconvert" software, which is the more appropriate constant to use.

This issue is also carried over in some models that carry "Dragen" in the name but likely should point to a more specific software.

Being more explicit might help us identify potential issues faster.

One examlpe can be found in this PR:
https://github.com/Clinical-Genomics/cg/pull/2019

## Suggested solution

Find the usages of Dragen and if applicable be explicit of the underlying software that is used when we parse information. 

## This can be closed when

*Describe what needs to be done for this issue to be closed*

## Blocked by

*If there are any blocking issues/prs/things in this or other repos. Please link to them.*

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Fixes #2030. Remove all ambiguity between the constants `BclConverter.DRAGEN` and `BclConverter.BCLCONVERT`, removing de former and replacing all the usages with the latter.

### Changed

- Removes `BclConverter.DRAGEN` and changes it with `BclConverter.BCLCONVERT`

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b deprecate-dragen-constant -a
    ```

### How to test

- [x] Do `cg demultiplex all` with flow cells that can be demultiplexed.

### Expected test outcome

- [x] Check that the demultiplex job starts
- [x] The demultiplexer finishes correctly

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by ChrOertlin — APPROVED** [review]
❤️

File: cg/constants/demultiplexing.py [hunk 1]
```diff
@@ -207,20 +206,20 @@ class IndexOverrideCycles(StrEnum):
...
     "--bcl-converter",
-    type=click.Choice(["bcl2fastq", "dragen"]),
+    type=click.Choice([BclConverter.BCL2FASTQ, BclConverter.BCLCONVERT]),
     default=None,
     help="Specify bcl conversion software. Choose between bcl2fastq and dragen. "
     "If not specified, the software will be determined automatically using the sequencer type.",
 )
 
 
 DEMUX_STATS_PATH: dict[str, dict[str, Path | None]] = {
```
[the remark this answers is not in the document]
Comment (diitaz93) [reply cg/constants/demultiplexing.py:216]: Yes, it is actually used for manual sample sheet creation, demultiplex and finishing flow cells. In the automation it is set to BCLConvert

**Comment by diitaz93** [comment]
## Tests on stage passed
Sample sheets created, demultiplex started correctly

**Comment by diitaz93** [comment]
## Deployed to stage and production

**Pull request merged**